tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/AgentMalfunctionTests/AgentMalfunctionTests.cs

```diff
@@ -0,0 +1,101 @@
+// <copyright file="AgentMalfunctionTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    [UsesVerify]
+    public class AgentMalfunctionTests : TestHelper
+    {
+        private static readonly Regex StackRegex = new(@"      error.stack:(\n|\r){1,2}.*(\n|\r){1,2}.*,(\r|\n){1,2}");
+        private static readonly Regex ErrorMsgRegex = new(@"      error.msg:.*,(\r|\n){1,2}");
+
+        public AgentMalfunctionTests(ITestOutputHelper output)
+            : base("ProcessStart", output)
+        {
+            SetServiceVersion("1.0.0");
+        }
+
+        [SkippableFact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SubmitsTracesWhenAgentDoesNotAnswer()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent();
+            agent.SetBehaviour(AgentBehaviour.NO_ANSWER);
+            TestInstrumentation(agent);
+        }
+
+        [SkippableFact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SubmitsTracesWhenAgentAnswersSlowly()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent();
+            agent.SetBehaviour(AgentBehaviour.SLOW_ANSWER);
+            TestInstrumentation(agent);
+        }
+
+        [SkippableFact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SubmitsTracesWhenAgentSendsWrongMessages()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent();
+            agent.SetBehaviour(AgentBehaviour.WRONG_ANSWER);
+            TestInstrumentation(agent);
+        }
+
+        [SkippableFact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SubmitsTracesWhenAgentReturns404()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent();
+            agent.SetBehaviour(AgentBehaviour.RETURN_404);
+            TestInstrumentation(agent);
+        }
+
+        [SkippableFact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SubmitsTracesWhenAgentReturns500()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent();
+            agent.SetBehaviour(AgentBehaviour.RETURN_500);
+            TestInstrumentation(agent);
+        }
+
+        private async void TestInstrumentation(MockTracerAgent agent)
```

Comment:
nit: You could write these as a single theory if you prefer 🙂 
```suggestion
        [SkippableTheory]
        [Trait("Category", "EndToEnd")]
        [Trait("RunOnWindows", "True")]
        [InlineData(AgentBehaviour.NO_ANSWER)
        [InlineData(AgentBehaviour.SLOW_ANSWER)
        [InlineData(AgentBehaviour.WRONG_ANSWER)
        [InlineData(AgentBehaviour.RETURN_404)
        [InlineData(AgentBehaviour.RETURN_500)
        public void SubmitsTraces(AgentBehaviour behaviour)
        {
            using var agent = EnvironmentHelper.GetMockAgent();
            agent.SetBehaviour(behaviour);
            TestInstrumentation(agent);
```

Reply:
Cool!